tiny_cnn/lossfunctions/loss_function.h

```diff
@@ -32,48 +32,67 @@ namespace tiny_cnn {
 // mean-squared-error loss function for regression
 class mse {
 public:
-    static float_t f(float_t y, float_t t) {
-        return (y - t) * (y - t) / 2;
+    static float_t f(const vec_t& y, const vec_t& t) {
+        assert(y.size() == t.size());
+        float_t d = 0.0;
+        for(unsigned int i = 0; i < y.size(); ++i)
+            d += (y[i] - t[i]) * (y[i] - t[i]);
+        return d/y.size();
     }
 
```

Comment:
@nyanp Ok, misunderstood your comment. Tests are still failing. Don't know what exactly happens in `generate_gradient_check_data` and `gradient_check`